sam/io_test.go

```diff
@@ -204,3 +204,90 @@ func TestReadAndWriteReal(t *testing.T) {
 		}
 	}
 }
+
+func TestGoReadToChanRecycleSam(t *testing.T) {
+	receiveChan, recycleChan, _ := GoReadToChanRecycle("testdata/small.sam", 1)
+	expected := []Sam{r001, r002, r003, r004, r003Supplemental, r001Supplemental}
+	var i int
+	for val := range receiveChan {
+		if !Equal(*val, expected[i]) {
```

Comment:
Jw, what is this Equal() coming from? Is it in the sam package and that's why we dont need a prefix like fileio.AreEqual()?